doc/cookbook/gpus.md

```diff
@@ -0,0 +1,226 @@
+
+An example pipeline definition for a GPU enabled Pachyderm Pipeline is as follows:
+
+```
+{
+  "pipeline": {
+    "name": "train"
+  },
+  "transform": {
+    "image": "acme/your-gpu-image",
+    "cmd": [ 
+      "python",
+      "train.py"
+    ],
+    "env": {
+      "LD_LIBRARY_PATH": "/usr/lib/nvidia:/usr/local/cuda/lib64:/rootfs/usr/lib/x86_64-linux-gnu"
+    }
+  },
+  "resource_spec": {
+      "gpu": 1
+  },
+  "inputs": {
+    "atom": {
+      "repo": "data",
+      "glob": "/*"
+    }
+  ]
+}
+```
+
+## Deploy a GPU Enabled Pachyderm Cluster
+
+**NOTE:** You can also [test Pachyderm + GPUs locally](#test-locally)
+
+**NOTE:** The following has been tested with these versions of k8s/kops:
+
+```
+$kubectl version
+Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.1", GitCommit:"b0b7a323cc5a4a2019b2e9520c21c7830b7f708e", GitTreeState:"clean", BuildDate:"2017-04-03T20:44:38Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
+Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:22:08Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
+$kops version
+Version 1.6.0-beta.1 (git-77f222d)
+```
+
+To deploy a Pachyderm cluster with GPU support we assume:
+
+- You're using kops for your deployment (which means you're using AWS or GCE, not GKE). Other deploy methods are available, but these are the ones we've tested most thoroughly. 
+- You have a working pachyderm cluster already up and running that you can connect to with `kubectl`.
+
+### Add GPU nodes to your k8s cluster
+
+For example, you can use `kops create ...`:
+
+```
+$kops create ig gpunodes --name XXXXXX-pachydermcluster.kubernetes.com --state s3://k8scom-state-store-pachyderm-YYYYY --subnet us-west-2c
+```
+
+where your specification will look something like:
+
+```
+  1 apiVersion: kops/v1alpha2
+  2 kind: InstanceGroup
+  3 metadata:
+  4   creationTimestamp: null
+  5   name: gpunodes
+  6 spec:
+  7   image: kope.io/k8s-1.5-debian-jessie-amd64-hvm-ebs-2017-01-09
+  8   machineType: p2.xlarge
+  9   maxSize: 1
+ 10   minSize: 1
+ 11   role: Node
+ 12   subnets:
+ 13   - us-west-2c
+```
+
+In this example we used Amazon's p2.xlarge instance which contains a single GPU node.
+
+### Enable GPUs at the k8s level
+
+Again, you can use `kops` to edit your cluster:
+
+```
+$kops edit cluster --name XXXXXXX-pachydermcluster.kubernetes.com --state s3://k8scom-state-store-pachyderm-YYYY
+```
+
+and add the fields:
+
+```
+  hooks:
+  - execContainer:
+      image: pachyderm/nvidia_driver_install:dcde76f919475a6585c9959b8ec41334b05103bb
+  kubelet:
+    featureGates:
+      Accelerators: "true"
+```
```

Comment:
Note - if this fails w a message that tells you the 'fields were not recognized' it means you need to upgrade kops (unless there was a typo). We should probably add this troubleshooting tip here.